azdev/operations/linter/linter.py

```diff
@@ -228,17 +259,35 @@ def run(self, run_params=None, run_commands=None, run_command_groups=None, run_h
 
     def _run_rules(self, rule_group):
         from colorama import Fore
-        for rule_name, (rule_func, linter_callable) in self._rules.get(rule_group).items():
+        for rule_name, (rule_func, linter_callable, rule_severity) in self._rules.get(rule_group).items():
+            severity_str = rule_severity.name
             # use new linter if needed
             with LinterScope(self, linter_callable):
-                violations = sorted(rule_func()) or []
-                if violations:
-                    print('- {} FAIL{}: {}'.format(Fore.RED, Fore.RESET, rule_name))
-                    for violation_msg in violations:
-                        print(violation_msg)
-                    print()
-                else:
-                    print('- {} pass{}: {} '.format(Fore.GREEN, Fore.RESET, rule_name))
+                # if the rule's severity is lower than the linter's severity skip it.
+                if self._linter_severity_is_applicable(rule_severity, rule_name):
+                    violations = sorted(rule_func()) or []
+                    if violations:
+                        if rule_severity == LinterSeverity.HIGH:
+                            sev_color = Fore.RED
+                        elif rule_severity == LinterSeverity.MEDIUM:
+                            sev_color = Fore.YELLOW
+                        else:
+                            sev_color = Fore.CYAN
+
+                        print('- {} FAIL{} - {}{}{} severity: {}'.format(Fore.RED, Fore.RESET, sev_color,
```

Comment:
Hello. Just want to know, will `print` directly show the log onto the screen? Do `Fore.RED` and `Fore.RESET` work like a switch?

Reply:
Is there a conclusion?

Reply:
yes.